src/FilterList.js

```diff
@@ -2,63 +2,91 @@ import React from 'react'
+import { Flex, Dynamic, Popover } from './layout'
+import { withStateLens } from './utils/mobx-react-utils'
 import InjectTreeNode from './utils/injectTreeNode'
 import DefaultIcon from './DefaultIcon'
 import { bdJoin } from './styles/generic'
 
 export let Label = inject(_.pick('tree'))(
-  observer(({ tree, node, Icon, ...x }) => (
-    <div
-      className={`filter-field-label ${
-        _.get('hasValue', node) ? 'filter-field-has-value' : ''
-      }`.trim()}
-      style={{
-        cursor: 'pointer',
-        display: 'flex',
-        justifyContent: 'space-between',
-      }}
-      onClick={() =>
-        tree && node && tree.mutate(node.path, { paused: !node.paused })
-      }
-    >
-      <span {...x} />
-      {tree &&
-        node && (
-          <Flex
-            className="filter-field-label-icon"
-            style={{ alignItems: 'center' }}
-          >
-            {!node.updating &&
-              tree.disableAutoUpdate &&
-              // find if any nodes in the tree are marked for update (i.e. usually nodes are marked for update because they react to "others" reactor)
-              _.some(
-                treeNode => treeNode !== node && treeNode.markedForUpdate,
-                F.treeToArray(_.get('children'))(tree.tree)
-              ) && (
-                <div
-                  className="filter-field-icon-refresh"
-                  onClick={e => {
-                    e.stopPropagation()
-                    tree.triggerUpdate()
+  withStateLens({ popover: false })(
+    observer(({ tree, node, Icon, ListItem: Item, popover, ...x }) => (
+      <Flex
+        className={`filter-field-label ${
+          _.get('hasValue', node) ? 'filter-field-has-value' : ''
+        }`.trim()}
+        style={{
+          cursor: 'pointer',
+          alignItems: 'center',
+          justifyContent: 'space-between',
+        }}
+        onClick={() =>
+          tree && node && tree.mutate(node.path, { paused: !node.paused })
+        }
+      >
+        <span {...x} />
+        {tree &&
+          node && (
+            <React.Fragment>
+              <span
+                onClick={e => {
+                  e.stopPropagation()
+                  F.flip(popover)()
+                }}
+              >
+                <Icon icon="TableColumnMenu" />
+                <Popover
+                  isOpen={popover}
+                  style={{
+                    userSelect: 'none',
+                    marginTop: '0.5rem',
+                    width: '5.5rem',
+                    transform: 'translateX(-2.25rem)',
+                    lineHeight: '1.4rem',
                   }}
                 >
-                  <Icon icon="Refresh" />
-                </div>
-              )}
-            <Icon
-              icon={node.paused ? 'FilterListExpand' : 'FilterListCollapse'}
-            />
-          </Flex>
-        )}
-    </div>
-  ))
+                  <Item onClick={() => tree.clear(node.path)}>
```

Comment:
This should probably show conditionally only when node.validate() returns a truthy value

Reply:
Can you elaborate more? Why don't we want to show the link if the node doesn't pass validation? Also, it'd be confusing to me at least if the button doesn't show up without some sort of indication of why it's missing (something like a tooltip saying that the node is borked).
Another thing is that actions call `dispatch`, which validates the node.

Reply:
When we validate a node it basically tells us if that node (filter) has a value and will participate in the search. So I was thinking that we should show a `Clear Filter` option if the filter has a value and will be part of the criteria for filtering data. If the filter/node has no value, clearing it has otherwise no effect really and seems redundant.

Reply:
I don't think this is a show stopper, just nice to have.

Reply:
I went with `node.hasValue`. This wouldn't be needed if contexture-client was smarter.

Reply:
Should we even show this menu item `Clear Filter` if the condition is not met? The delete option in the other PR you have makes sense to always show until the actual node is removed from the tree but this one should only show if `node.hasValue` is true.

Reply:
Done. One thing though: right now we show an empty popover but this won't be an issue once we've got a link for deleting a filter there.